### ISSUE TITLE
UNet cannot be saved in `diffusers` nor `diffusers_safetensors` format

### ISSUE THREAD

**Issue by nickkolok** [kohya-ss/sd-scripts#966]
Hi,
I was running the script on Google Colab via the following command:
```bash
!accelerate launch --num_cpu_threads_per_process 1 sd-scripts/train_db.py \
    --pretrained_model_name_or_path=$MODEL_NAME \
    --dataset_config=$repo_path_conc/initial.toml \
    --output_dir=$OUTPUT_DIR  \
    --save_model_as=diffusers \
    --prior_loss_weight=1.0 \
    --max_train_steps=$SAVE_MAX_STEPS \
    --save_every_n_steps=$SAVE_INTERVAL \
    --learning_rate=3.4e-7 \
    --lr_scheduler="cosine" \
    --optimizer_type="AdamW8bit" \
    --xformers \
    --mixed_precision="fp16" \
    --gradient_checkpointing \
    --sample_every_n_steps=$SAVE_INTERVAL \
    --sample_sampler="k_euler" \
    --sample_prompts="/content/prompts-initial.txt"
```
Everything went fine, the samples became more and more pretty, the loss found its minimum... but at the end of training I noticed absence of the main part - UNet files!

![image](https://github.com/kohya-ss/sd-scripts/assets/5008131/dda9224b-724c-4cf2-9d26-9daab1a7729c)

What am I doing wrong?

**Comment by nickkolok** [comment]
Toggling to `--save_model_as=diffusers_safetensors` did not help. I noticed the huge error message (sorry for not providing it before) that ends with:
```
  (conv_norm_out): GroupNorm(32, 320, eps=1e-05, affine=True)
  (conv_act): SiLU()
  (conv_out): Conv2d(320, 4, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
) of type <class 'library.original_unet.UNet2DConditionModel'> cannot be saved.
model saved.
```

**Comment by nickkolok** [comment]
I've added `--output_name="pytorch_model" `  to the script arguments - still no effect. Keep going!

**Comment by nickkolok** [comment]
Both `--save_model_as=ckpt` and `--save_model_as=safetensors` produce files of 3.82 GB (according to Google Colab UI). Looks like a bug!

**Comment by nickkolok** [comment]
Please find the error log here:
[save-log.txt](https://github.com/kohya-ss/sd-scripts/files/13441401/save-log.txt)

Switching from `diffusers==0.21.2` (as specified in `requirements.txt`) to `diffusers==0.23.1` did not solve the problem.

**Comment by nickkolok** [comment]
With `diffusers==0.10.1`, the script failed to launch.
`0.12.1` - the same error as with  `0.10.1`
`0.16.1` - the same error again
(no intermediate versions of `diffusers` found)
`0.17.0` - no UNet saved
`0.17.1` - no UNet saved
`0.18.1` - no UNet saved
`0.19.1` - no UNet saved
`0.20.1` - no UNet saved
`0.21.2` - no UNet saved (as stated above)
`0.23.1` - no UNet saved (as stated above)

**Comment by nickkolok** [comment]
UPD: on `sd-scripts` checkouted to tag `v0.6.6` there seems to be no such problem

**Comment by quocanh34** [comment]
Have you fixed this bug?

**Comment by unko931** [comment]
@nickkolok 
I encountered this problem during `fine_tune.py`, it seems to have been resolved in `v0.8.1`.
